### PR TITLE
Update datatables

### PR DESCRIPTION
**Pull request by yubiuser**
**What does this PR aim to accomplish?:**

1) Updates `datatables`:

- `datatables` to `1.13.8`
- `datatables` plugin `button` to `2.4.2`
- ~~`datatables` plugin `select` to `1.7.0`~~ (was reverted during the development of this PR: the double check mark issue could not be solved)

2) Fixes a warning reported on the network table

> DataTables warning: DataTables warning: table id=network-entries - Requested unknown parameter 'ip' for row 0, column 1. For more information about this error, please see https://datatables.net/tn/4

`/network/device` won't return `ip` as top level key, but

```
      "macVendor": "Google, Inc.",
      "ips": [
        {
          "ip": "10.0.1.57",
          "name": null,
          "lastSeen": 1700344560,
          "nameUpdated": 1700344560
        }
```

Datatables data source was changed accordingly.

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

**Comment by yubiuser** [comment]
Needs a bit of care
Screenshot at 2023-11-18 23-14-40
Screenshot at 2023-11-18 23-15-02

File: scripts/pi-hole/js/groups-clients.js [hunk 1]
```diff
@@ -100,6 +100,7 @@ function initTable() {
       dataSrc: "clients",
       type: "GET",
     },
+    pagingTag: "button",
```
Comment (rdwebdesign) [inline scripts/pi-hole/js/groups-clients.js]: After adding the bootstrap3 integration, this seems to be unnecessary.

**Review by rdwebdesign — DISMISSED** [review]

**Comment by DL6ER** [comment]
Something isn't right with colors:

Unselected:
Screenshot from 2023-11-19 07-30-37

Selected on `development-v6`:
Screenshot from 2023-11-19 07-30-47

---

So far, so good, but:

Selected on this branch (latest commit by @rdwebdesign):
Screenshot from 2023-11-19 07-30-56

Selected on @yubiuser's last commit (https://github.com/pi-hole/web/pull/2852/commits/a8bb9ed201879864c6386ba2334d07ac543516ec):
grafik

**Comment by rdwebdesign** [comment]
~~Hmmm... I forgot to add the CSS variables on the other themes.~~

EDIT:

Actually, the same is happening in all themes, but it's an issue in dark themes (the link color is dark).
This issue is related to some changes in datatables styles. 
Now they use a few CSS variables:
```css
:root {
	--dt-row-selected: 0, 136, 204;
	--dt-row-selected-text: 255, 255, 255;
	--dt-row-selected-link: 9, 10, 11;
	--dt-row-stripe: 0, 0, 0;
	--dt-row-hover: 0, 0, 0;
	--dt-column-ordering: 0, 0, 0;
	--dt-html-background: white;
}
```
But this is also changing the check mark to white, when the row is selected.

I will finish it later.

File: scripts/pi-hole/js/groups-clients.js (hunk 1 above)
Comment (yubiuser) [reply scripts/pi-hole/js/groups-clients.js]: I reverted the commit.

**Review by DL6ER — APPROVED** [review]
It seems to work everywhere I looked

**Comment by sefinek** [comment]
brave_BhJbAZBh3JLD


https://github.com/pi-hole/web/assets/92880265/bbcbaa48-06ec-4bc3-9ae2-1494e24d9b8f

**Comment by DL6ER** [comment]
Thanks for the report. We need to remove `<ul>` + `</ul>`

https://github.com/pi-hole/web/blob/4d3d64b90426fc97eb6dc0425b33ed8956f71c14/scripts/pi-hole/js/groups-domains.js#L444

and friends in similar files to fix this

**Comment by Th3M3** [comment]
Looks like the processing indicator also needs some patch to not just show a gray box.

emptyanim

**Comment by rdwebdesign** [comment]
Ohh... I know what is the issue here. 

The updated datatables version introduced a new custom property: `--dt-row-selected`

This property accepts RGB colors _(actually, it just the numbers, not a real color code... this property looks strange)_, but I used a color in hex format.

**Comment by yubiuser** [comment]
The updated datatables caused to much trouble and needed to many manual changes to your styles. This PR https://github.com/pi-hole/web/pull/2874 is reverting the changes. At one point we should update out web base (adminlte) and can re-apply updates to all components

**Pull request merged**